models/transmission.go

```diff
@@ -31,32 +32,44 @@ type Transmission struct {
 	isValidated  bool
 }
 
-// Custom marshaling to make empty strings null
+// Marshal returns a JSON encoded byte array representation of the model
```

Comment:
Can this be simplified in a similar manner to https://github.com/edgexfoundry/go-mod-core-contracts/pull/163#pullrequestreview-299392862?

i.e.

```
type transmissionCommon struct {
	Timestamps
	ID           string               `json:"id,omitempty"`
	Receiver     string               `json:"receiver,omitempty"`
	Status       TransmissionStatus   `json:"status,omitempty"`
	Records      []TransmissionRecord `json:"records,omitempty"`
}

type transmissionCommonAlias transmissionCommon

type Transmission struct {
	transmissionCommon
	Notification Notification         `json:"notification"`
	Channel      Channel              `json:"channel,omitempty"`
	ResendCount  int                  `json:"resendcount"`
	isValidated  bool
}

type transmissionAlias Transmission

// Marshal returns a JSON encoded byte array representation of the model
func (t Transmission) MarshalJSON() ([]byte, error) {
	alias := struct {
		transmissionCommonAlias
		Notification *Notification         `json:"notification,omitempty"`
		Channel      *Channel              `json:"channel,omitempty"`
		ResendCount  *int                  `json:"resendcount,omitempty"`
	}{
		transmissionCommonAlias: transmissionCommonAlias(t.transmissionCommon),
	}
	if !reflect.DeepEqual(t, Transmission{}) {
		alias.ResendCount = &t.ResendCount
	}
	// do not marshal empty member objects
	if !reflect.DeepEqual(t.Notification, Notification{}) {
		alias.Notification = &t.Notification
	}
	if !reflect.DeepEqual(t.Channel, Channel{}) {
		alias.Channel = &t.Channel
	}
	return json.Marshal(alias)
}

// UnmarshalJSON implements the Unmarshaler interface for the Transmission type
func (t *Transmission) UnmarshalJSON(data []byte) error {
	var err error
	var alias transmissionAlias
	if err = json.Unmarshal(data, &alias); err != nil {
		return err
	}
	*t = Transmission(alias)
	t.isValidated, err = t.Validate()
	return err
}
```

Note impact on corresponding tests...

```
var TestTransmission = Transmission{
	transmissionCommon: transmissionCommon{
		Timestamps: testTimestamps,
		Receiver: "test receiver",
		Status: TransmissionStatus(Sent),
		Records: []TransmissionRecord{TestSentTransRecord},
	},
	Notification: TestNotification,
	Channel: Channel{
		Type: ChannelType(Email),
		MailAddresses: []string{"jpwhite_mn@yahoo.com", "james_white2@dell.com"},
	},
	ResendCount: 0,
}
...
		{"test string of transmission", &TestTransmission, "{\"created\":123,\"modified\":123,\"origin\":123,\"receiver\":\"test receiver\",\"status\":\"SENT\",\"records\":[{\"status\":\"SENT\",\"response\":\"ok\",\"sent\":123}],\"notification\":{\"created\":123,\"modified\":123,\"id\":\"" + TestNotificationID + "\",\"slug\":\"test slug\",\"sender\":\"test sender\",\"category\":\"SECURITY\",\"severity\":\"CRITICAL\",\"content\":\"test content\",\"description\":\"test description\",\"status\":\"NEW\",\"labels\":[\"label1\",\"labe2\"],\"contenttype\":\"text/plain\"},\"channel\":{\"type\":\"EMAIL\",\"mailAddresses\":[\"jpwhite_mn@yahoo.com\",\"james_white2@dell.com\"]},\"resendcount\":0}"},

```

This approach eliminates as much struct field duplication as possible and gets rid of the hack returning "{}" string literal.

Only calling out this instance; there may be others.